package_nso_to_oc/xe/xe_routing_policy.py

```diff
@@ -0,0 +1,521 @@
+NSO_HOST - IP address or hostname for the NSO server
+NSO_USERNAME
+NSO_PASSWORD
+NSO_DEVICE - NSO device name for configuration translation
+TEST - True or False. True enables sending the OpenConfig to the NSO server after generation
+"""
+
+import sys
+from importlib.util import find_spec
+import re
+
+routing_policy_notes = []
+openconfig_routing_policies = {
+    "openconfig-routing-policy:routing-policy": {
+        "openconfig-routing-policy:defined-sets": {
+            "openconfig-bgp-policy:bgp-defined-sets": {},
+            "openconfig-routing-policy:tag-sets": {
+                "openconfig-routing-policy:tag-set": []
+            }
+        },
+        "openconfig-routing-policy:policy-definitions": {
+            "openconfig-routing-policy:policy-definition": []
+        }
+    }
+}
+# OC has an additional "NOPEER" BGP community member, which is not support in XE.
+well_known_members = {
+    "no-export": "NO_EXPORT",
+    "no-advertise": "NO_ADVERTISE",
+    "local-as": "NO_EXPORT_SUBCONFED"
+}
+policy_results = {
+    "permit": "ACCEPT_ROUTE",
+    "deny": "REJECT_ROUTE"
+}
+port_operators = ["range", "eq", "lt", "gt", "neq"]
+regex_ipv4_masklength_range = re.compile(r'([0-9]{1,2})\.\.([0-9]{1,2})')
+regex_meta = {'[', '\\', '.', '^', '$', '*', '+', '?', '{', '|', '('}
+INLINE = "INLINE"
+REFERENCE = "REFERENCE"
+ADD = "ADD"
+REPLACE = "REPLACE"
+REMOVE = "REMOVE"
+ACL_STD_TYPE = "ACL_IPV4_STANDARD"
+ACL_EXT_TYPE = "ACL_IPV4"
+
+def xe_routing_policy(config_before, config_after):
+    process_prefix_sets(config_before, config_after)
+    process_as_path_sets(config_before, config_after)
+    process_community_sets(config_before, config_after)
+    process_ext_community_sets(config_before, config_after)
+    process_policy_definitions(config_before, config_after)
+
+def process_prefix_sets(config_before, config_after):
+    prefix_sets = {"openconfig-routing-policy:prefix-sets": {"openconfig-routing-policy:prefix-set": []}}
+    xe_prefixes = config_before.get("tailf-ned-cisco-ios:ip", {}).get("prefix-list", {}).get("prefixes", [])
+    xe_prefixes_after = config_after.get("tailf-ned-cisco-ios:ip", {}).get("prefix-list", {}).get("prefixes", [])
+
+    for prefix_index, prefix in enumerate(xe_prefixes):
+        new_prefix_set = {
+            "openconfig-routing-policy:name": prefix.get("name"),
+            "openconfig-routing-policy:config": {
+                "openconfig-routing-policy:name": prefix.get("name"),
+                "openconfig-routing-policy:mode": "IPV4"
+            },
+            "openconfig-routing-policy:prefixes": {"openconfig-routing-policy:prefix": []}
+        }
+        prefixes = new_prefix_set["openconfig-routing-policy:prefixes"]["openconfig-routing-policy:prefix"]
+        seq_list_after = common.get_index_or_default(xe_prefixes_after, prefix_index, {}).get("seq", [])
+        all_processed = True
+
+        for seq_index, seq in enumerate(prefix.get("seq", [])):
+            if "deny" in seq:
```

Comment:
If deny in prefix-list, we must note, keep processing for other notes, but prefix-list missing statements can't be appended to OC prefix lists.